nDPI-patch/README.md

```diff
@@ -196,3 +196,33 @@ to
 #define assert(x)
 #endif
 ```
+
+### Kernel Version Mismatch
+
+Between versions 5.4 and 5.8, the Linux Kernel changed how it keeps
+track of time. `nDPI` has 5.4 assumptions baked in.  If you see an error like
+```
+unkown type name 'time_t'
+```
+during the build process, this may mean `time_t` is no longer supported 
+and you will need to rollback. You should also see a line like
+```
+make[2]: Leaving directory '/usr/src/linux-headers-5.8.0-1040-generic'
+```
+giving you insight on which kernel version you are building with.
+
+To rollback:
+1. Install 5.4 kernel then reboot
+    ```
+    $ sudo apt install linux-generic
+    $ reboot
+    ```
+1. Log back in and list the kernel packages
+    ```
+    $ dpkg --list | grep linux
+    ```
+1. Remove 5.8 headers, modules, modules-extra packages and reboot again
```

Comment:
5.8 is likely the most common wrong version but it might be 5.6 or, later, 5.9.

I'd suggest revising the `dpkg` command to grep for the version identified earlier then this could just say remove the packages for the other kernel version.

Reply:
That's a good call.